### ISSUE TITLE
where is self.loss coming from? not able to find ... plz help

### ISSUE THREAD

**Issue by bkumardevan07** [as-ideas/TransformerTTS#58]
in weighted_sum_losses .....
`
def _gta_forward(self, inp, tar, stop_prob, training):
        tar_inp = tar[:, :-1]
        tar_real = tar[:, 1:]
        tar_stop_prob = stop_prob[:, 1:]
        
        mel_len = int(tf.shape(tar_inp)[1])
        tar_mel = tar_inp[:, 0::self.r, :]
        
        with tf.GradientTape() as tape:
            model_out = self.__call__(inputs=inp,
                                      targets=tar_mel,
                                      training=training)
            loss, loss_vals = weighted_sum_losses((tar_real,
                                                   tar_stop_prob,
                                                   tar_real),
                                                  (model_out['final_output'][:, :mel_len, :],
                                                   model_out['stop_prob'][:, :mel_len, :],
                                                   model_out['mel_linear'][:, :mel_len, :]),
                                                  **self.loss**, # this 
                                                  self.loss_weights)
        model_out.update({'loss': loss})
        model_out.update({'losses': {'output': loss_vals[0], 'stop_prob': loss_vals[1], 'mel_linear': loss_vals[2]}})
        model_out.update({'reduced_target': tar_mel})
        return model_out, tape
`

**Comment by cfrancesco** [comment]
Hi,
this is created in the background by tensorflow's compile function.

**Comment by bkumardevan07** [comment]
is your tts loss average of (absolute error, cross entropy, mean squared error...) ???
Can you tell me what is actual tts loss used in literature. Actual I was looking but unable to find.
Thanks
